docs/repo_structure.md

```diff
@@ -0,0 +1,73 @@
+# Repository structure
+
+
+### Rpm-ostree source code
+```
+.
+└─ src                          Contains source code for rpm-ostree
+```
+
+#### File structures under src
+```
+.
+└─ src
+  ├── app                       Contains the "frontend" logic (display CLI commands, function calls to daemon etc) for each command
+  |
+  ├── daemon                    Usually has the "backend" logic for each command that involves with dbus daemon and transactions
+  |
+  ├── lib                       Contains package and db libraries
```

Comment:
How about something like "Public library; contains APIs for exploring rpmdb in OSTrees" ?